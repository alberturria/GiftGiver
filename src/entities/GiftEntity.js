export default class GiftEntity{
    constructor(id, person, present) {
        this.id = id;
        this.person = person;
        this.present = present;
    }
}