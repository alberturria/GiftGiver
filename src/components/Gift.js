import React, { Component } from 'react';
import GiftEntity from '../entities/GiftEntity';
import { Form, FormGroup, FormControl, Button, FormLabel } from 'react-bootstrap';

export default class Gift extends Component {
    constructor() {
        super();
        this.state = { gift: new GiftEntity(null, '', '') }
    }

    _handleChangePerson = (event) => {
        const { gift } = this.state;
        gift.person = event.target.value;

        this.setState({ gift });
    }

    render() {
        return (
        <div>
            <Form>
                <FormGroup>
                    <FormLabel>Person</FormLabel>
                    <FormControl
                    className='input-person'
                    onChange={this._handleChangePerson}
                    />
                </FormGroup>
            </Form>
        </div>
            );
    }

}