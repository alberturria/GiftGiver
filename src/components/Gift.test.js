import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift.js';
import GiftEntity from '../entities/GiftEntity.js';

describe('Gift', () => {
    const gift = shallow(<Gift />);

    it('Should\'ve rendered a Gift component', () =>{
        expect(gift).toMatchSnapshot();
    })

    it('Should initialize a person and present in the state', () =>{
        const expectedGift = new GiftEntity(null, '', '');
        expect(gift.state().gift).toEqual(expectedGift);
    })

    describe('When typing into the person input', () => {
        const samplePerson = 'SamplePerson';
        beforeEach(() => {
            const inputPerson = gift.find('.input-person');
            inputPerson.simulate('change', { target: { value: samplePerson }});
        })

        it('Should update the person in the state', () => {
            expect(gift.state().gift.person).toEqual(samplePerson);

        });

    });
});