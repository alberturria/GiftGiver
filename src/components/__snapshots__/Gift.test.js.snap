// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Should've rendered a Gift component 1`] = `
<div>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        className="input-person"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
</div>
`;
